keanu-project/src/test/java/io/improbable/keanu/tensor/dbl/Nd4jBroadcastTensorTest.java

```diff
@@ -0,0 +1,513 @@
+          b = np.array([1, 2, 3, 4]).reshape(2,2,1,1,1)
+          ab = a * b
+          print(ab)
+          print(np.shape(ab))
+         */
+
+        DoubleTensor rank5 = Nd4jDoubleTensor.create(new double[]{
+            1, 2, 3, 4, 5, 6, 7, 8, 4, 3, 2, 1, 7, 5, 8, 6,
+            6, 3, 2, 9, 3, 4, 7, 6, 6, 2, 5, 4, 0, 2, 1, 3
+        }, new int[]{2, 2, 2, 2, 2});
+
+        DoubleTensor matrix = matrixA.reshape(2, 2, 1, 1, 1);
+        DoubleTensor expected = Nd4jDoubleTensor.create(new double[]{
+            1, 2, 3, 4, 5, 6, 7, 8, 8, 6, 4, 2, 14, 10, 16, 12,
+            18, 9, 6, 27, 9, 12, 21, 18, 24, 8, 20, 16, 0, 8, 4, 12
+        }, new int[]{2, 2, 2, 2, 2});
+
+        assertTimesOperationEquals(rank5, matrix, expected);
+        assertTimesInPlaceOperationEquals(rank5, matrix, expected);
+    }
+
+    @Test
+    public void canSuperBroadcast() {
+
+        /*
+          a = np.zeros([2,2,2,2])
+          b = np.array([1,0,1,0]).reshape(2,2)
+          ab = a + b
+          print(ab)
+          print(np.shape(ab))
+         */
+
+        DoubleTensor x = Nd4jDoubleTensor.zeros(new int[]{2, 2, 2, 2});
+        DoubleTensor y = Nd4jDoubleTensor.create(new double[]{1, 0, 1, 0}, new int[]{2, 2});
+
+        DoubleTensor diff = x.plus(y);
+
+        DoubleTensor expected = Nd4jDoubleTensor.create(new double[]{
+            1, 0, 1, 0,
+            1, 0, 1, 0,
+            1, 0, 1, 0,
+            1, 0, 1, 0
+        }, new int[]{2, 2, 2, 2});
+
+        assertEquals(expected, diff);
+    }
+
+    @Test
+    public void canSuperBroadcastInPlace() {
+        DoubleTensor x = Nd4jDoubleTensor.zeros(new int[]{2, 2, 2, 2});
+        DoubleTensor y = Nd4jDoubleTensor.create(new double[]{1, 0, 1, 0}, new int[]{2, 2});
+
+        DoubleTensor diff = x.plusInPlace(y);
+
+        DoubleTensor expected = Nd4jDoubleTensor.create(new double[]{
+            1, 0, 1, 0,
+            1, 0, 1, 0,
+            1, 0, 1, 0,
+            1, 0, 1, 0
+        }, new int[]{2, 2, 2, 2});
+
+        assertEquals(expected, diff);
+    }
+
+    @Test
+    public void canBroadcastAdd() {
+
+        /*
+          x = np.array([1,2,3]).reshape(3,1)
+          s = np.array([-5, -2, -3, -7, -8, -5, -2, -3, -7, -8, -5, -2, -3, -7, -8]).reshape(3,5)
+          sx = s + x
+          print(sx)
+          print(np.shape(sx))
+         */
+
+        DoubleTensor x = Nd4jDoubleTensor.create(new double[]{1, 2, 3}, new int[]{3, 1});
+        DoubleTensor s = Nd4jDoubleTensor.create(new double[]{
+            -5, -2, -3, -7, -8,
+            -5, -2, -3, -7, -8,
+            -5, -2, -3, -7, -8
+        }, new int[]{3, 5});
+
+        DoubleTensor diff = s.plus(x);
+        DoubleTensor expected = Nd4jDoubleTensor.create(new double[]{
+            -4, -1, -2, -6, -7,
+            -3, 0, -1, -5, -6,
+            -2, 1, 0, -4, -5
+        }, new int[]{3, 5});
+
+        assertEquals(expected, diff);
+    }
+
+    @Test(expected = AssertionError.class)
```

Comment:
Should this be `@Ignore`? http://junit.sourceforge.net/javadoc/org/junit/Ignore.html